### PR TITLE
Storm weather related fixes

### LINKED ISSUE #76304: Ash storm sound loops regardless of when the storm is happening
## Reproduction:
So lately i've noticed that the beginning sound of the ashstorm on lavaland seems to play regardless of whether or not a storm is happening. Hitting stop sounds doesn't seem to work at all.

### LINKED ISSUE #76400: Nightmares still take damage from lavaland ashstorms during shadow jaunt
RoundID 209278

be nightmare, go to lavaland during storm, jaunt, still take damage from storm despite being in darkness and in shadow dimension

### PR DESCRIPTION
**Pull request by Sealed101**
## About The Pull Request
A bundle of stormy weather-related fixes:
- ash storm visually replenishes dug basalt again; also updates refilled basalt's light, if it rolled an eligible icon variant 
  - new proc refill_dug() for asteroid turfs, currently made use of in weather but be my guest
  - I THINK the lowest-light variants are so low they don't show up at all but I didn't look into the lighting subsystem so /shrug
  - `set_basalt_light()` is a global proc. the reason is because we have, like, an unnecessary amount of basalt turfs that don't call back to each other. `misc/basalt`, `floor/fakebasalt`, `holofloor/basalt`. bro come on. should merge them where possible but that's for a separate PR; ALSO MAKE IT USE THE `diggable` ELEMENT BRO COME ON!!!
- ash storm no longer loops `end()` sounds
  - it was calling parent THEN removing the playlist's sounds, which caused the playlist's ash storm weak sound to be picked again and play with a hugelarge timer
- snow storm no longer affects mecha pilots
  - I find it weird that ash storm immunity is just innate whether the mech is "sealed" or not (Ripley mk I) but for now, I've made snow storm immunity behave the same
- _jaunt spells provide weather immunity while the jaunt is active_
  - _the caster is supposed to be untouchable by any practical means, it don't make a lick'a sense to have weather affect them. I gave it a general weather immunity however, and that also makes jaunts protect from radiation. methinks it's fair, lmk if that's excessive_

## Why It's Good For The Game
- Fixes #76304
- Fixes #76400
- Refilled basalt tiles that should have light are consistent with mapped in basalt tiles
- Consistency for mechs; the adjustment might be a balance thing, lmk

## Changelog
:cl:
fix: fixed basalt turfs remaining visually dug up when refilled by an ash storm
fix: fixed ash storm ending sounds looping after the storm is over
fix: mechs are now snow storm immune
fix: jaunt spells protect from weather when jaunting
/:cl:

### CODE REVIEW

**Review by Jacquerel — APPROVED** [review]

**Comment by LemonInTheDark** [comment]
> ALSO MAKE IT USE THE diggable ELEMENT BRO COME ON!!!

Likely too common, large init times hit for little upside, tho it might be worth testing

**Comment by Sealed101** [comment]
>Likely too common, large init times hit for little upside, tho it might be worth testing

i guess, but the element is being used for those previously mentioned excess types of the basalt/snow turfs so 🤷 

`misc/basalt`, `misc/snow`, `floor/grass`, `floor/fake_snow` and `floor/fakebasalt` to be exact, and they are of dubious use either way

File: code/modules/vehicles/mecha/_mecha.dm [hunk 1]
```diff
@@ -257,6 +257,7 @@
 	AddElement(/datum/element/atmos_sensitive, mapload)
 	become_hearing_sensitive(trait_source = ROUNDSTART_TRAIT)
 	ADD_TRAIT(src, TRAIT_ASHSTORM_IMMUNE, ROUNDSTART_TRAIT) //protects pilots from ashstorms.
+	ADD_TRAIT(src, TRAIT_SNOWSTORM_IMMUNE, ROUNDSTART_TRAIT) //protects pilots from snowstorms
```
Comment (san7890) [inline code/modules/vehicles/mecha/_mecha.dm:260]: should use `add_traits()` proc here instead

Comment (Sealed101) [reply code/modules/vehicles/mecha/_mecha.dm]: https://github.com/tgstation/tgstation/pull/76367/commits/a65c54380b79856161f042425225bd1d54627e4c

**Comment by Sealed101** [comment]
jaunt spell changes, lmk if I've missed some jaunt spell/gone overboard. PR body updated

**Comment by Sealed101** [comment]
looking into #75352

**Comment by Sealed101** [comment]
can't exactly pin that issue down. it seems like it should be handled in a separate PR

**Pull request merged**